### PR TITLE
ICMPv6 cleanup

### PR DESCRIPTION
**Pull request by rveerama1**
Part of ICMPv6 modified as per https://github.com/zephyrproject-rtos/zephyr/issues/8721.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

**Review by pfalcon — CHANGES_REQUESTED** [review]
I once criticized these "fast path" cases, because they complicate situation with unit testing and code coverage - we unlikely have (enough) testcases to test slow path. Of course, it's pretty simple concept, so should "just work", but who knows if in one of the cases a typo or thinko is made. It's also hard to assess its impact on the IP stack code size (except that it's fair to say that Zephyr's stack sis not the lightest on the scene).

So, overall, it was hard to shake off the feeling that this fast path/slow path design was a premature optimization in our case, when the stack isn't really stable and more code paths don't really help with that.

So, I won't regret seeing it go, especially if @jukkar and @tbursztyka think the same. And I write all these details, in the hope that this is not just a random change for icmpv6 module, but will be propagated thru the entire stack. When we have a perfectalized stack, we can reintroduce that optimization, under code size and coverage control.

However, I spotted an inconsistency during conversion, so let me request a fix (or explanation why the discrepancy).

File: subsys/net/ip/icmpv6.c [hunk 1]
```diff
@@ -328,60 +281,37 @@ struct net_icmpv6_na_hdr *net_icmpv6_get_na_hdr(struct net_pkt *pkt,
...
 		NET_ERR("Cannot set the ICMPv6 NA header");
-		return NULL;
+		return -EINVAL;
 	}
 
-	return hdr;
+	return 0;
 }
 
 struct net_icmpv6_ra_hdr *net_icmpv6_get_ra_hdr(struct net_pkt *pkt,
```
[the remark this answers is not in the document]
Comment (rveerama1) [reply subsys/net/ip/icmpv6.c]: Yes, I modified _get_xxx() functions also.

**Comment by rveerama1** [comment]
Updated according to @pfalcon comments.

**Review by pfalcon — APPROVED** [review]
Thanks.

**Pull request merged**